openvm/src/powdr_extension/chip.rs

```diff
@@ -12,52 +9,48 @@ use crate::{
 
 use super::{executor::PowdrExecutor, opcode::PowdrOpcode, PowdrPrecompile};
 use itertools::Itertools;
-use openvm_circuit::system::memory::MemoryController;
-use openvm_circuit::{
-    arch::{ExecutionState, InstructionExecutor, Result as ExecutionResult},
-    system::memory::OfflineMemory,
-};
-use openvm_instructions::{instruction::Instruction, LocalOpcode};
+use openvm_instructions::LocalOpcode;
 use openvm_stark_backend::{
     p3_air::{Air, BaseAir},
-    rap::ColumnsAir,
+    p3_matrix::dense::DenseMatrix,
+    prover::{hal::ProverBackend, types::AirProvingContext},
+    ChipUsageGetter,
 };
 
 use openvm_stark_backend::{
-    config::{StarkGenericConfig, Val},
     interaction::InteractionBuilder,
     p3_field::PrimeField32,
     p3_matrix::Matrix,
-    prover::types::AirProofInput,
     rap::{AnyRap, BaseAirWithPublicValues, PartitionedBaseAir},
-    Chip, ChipUsageGetter,
+    Chip,
 };
+use openvm_stark_sdk::p3_baby_bear::BabyBear;
 use powdr_autoprecompiles::{
     expression::{AlgebraicEvaluator, AlgebraicReference, WitnessEvaluator},
     Apc,
 };
 
-pub struct PowdrChip<F: PrimeField32> {
+pub struct PowdrChip {
     pub name: String,
     pub opcode: PowdrOpcode,
     /// An "executor" for this chip, based on the original instructions in the basic block
-    pub executor: PowdrExecutor<F>,
-    pub air: Arc<PowdrAir<F>>,
+    pub executor: PowdrExecutor,
+    pub air: Arc<PowdrAir<BabyBear>>,
 }
```

Comment:
No more generic over F because `PowdrExecutor` is now `BabyBearSC` specific, and we can't get an `SC` from `F`.